src/target/register.c

```diff
@@ -93,9 +93,8 @@ void register_unlink_cache(struct reg_cache **cache_p, const struct reg_cache *c
 /** Marks the contents of the register cache as invalid (and clean). */
 void register_cache_invalidate(struct reg_cache *cache)
 {
-	struct reg *reg = cache->reg_list;
-
-	for (unsigned int n = cache->num_regs; n != 0; n--, reg++) {
+	for (size_t n = 0; n < cache->num_regs; n++) {
+		struct reg *reg = &cache->reg_list[n];
```

Comment:
Can you leave this as-is? I'd rather not make unnecessary target-independent changes in this branch, which has already drifted significantly from upstream.

Reply:
Yes, you are right, in the meantime I published an MR to the upstream about this.